packages/ui-components/src/components/text-field/text-field.tsx

```diff
@@ -0,0 +1,74 @@
+import { Component, Event, EventEmitter, h, Host, Prop, State, Watch } from '@stencil/core';
+import { EInputFieldType } from './text-field.types';
+
+@Component({
+	tag: 'kv-text-field',
+	styleUrl: 'text-field.scss',
+	shadow: true
+})
+export class KvTextField {
+	/** (optional) Text field tyoe */
+	@Prop({ reflect: true }) type!: EInputFieldType;
+	/** (optional) Text field label */
+	@Prop({ reflect: true }) label: string;
+	/** (optional) Text field input name */
+	@Prop({ reflect: true }) inputName: string;
+	/** (optional) Text field place holder */
+	@Prop({ reflect: true }) placeholder: string;
+	/** (optional) Text field disabled */
+	@Prop({ reflect: true }) disabled = false;
+	/** (optional) Text field required */
+	@Prop({ reflect: true }) required = false;
+	/** (optional) Text field loading state */
+	@Prop({ reflect: true }) loading = false;
+	/** (optional) Text field help text */
+	@Prop({ reflect: true }) helpText: string;
+	/** (optional) Text field slim style */
+	@Prop({ reflect: true }) slim = false;
+
+	/** Text field value */
+	@Prop({ reflect: true, mutable: true }) value: string;
+
+	/** Whether the state is ON or `true` */
+	@State() _value: string;
+
+	/** Watch `value` property for changes and update `_value` accordingly */
+	@Watch('value')
+	stateHandler(newValue: string) {
+		this._value = newValue;
+	}
+
+	/** Emitted when text field's value changes */
+	@Event() textChange: EventEmitter<InputEvent>;
+
+	private handleChange(event) {
+		this.value = event.target.value;
+		this.textChange.emit(event);
+	}
+
+	render() {
+		return (
+			<Host>
+				<div class="text-field-container">
+					{this.required && <span class="error">*</span>}
+					{this.label && <span class="label">{this.label}</span>}
+					<div class="input-container">
+						{!this.loading && (
+							<input
+								type={this.type}
+								name={this.inputName}
+								placeholder={this.placeholder}
+								disabled={this.disabled}
+								value={this._value}
+								onInput={event => this.handleChange(event)}
```

Comment:
```tsx
onInput={this.handleChange}
class={{'kv-slim-text-field': this.slim}}
```

without the change to your onInput handler, the template will create function every time and refer to the handler instead of referring directly to the function.

please add the blur event emitter again

Reply:
I removed the resolved flag on this one, use this class assignment approach please.